schema/helm-testsuite.json

```diff
@@ -0,0 +1,668 @@
+            "markdownDescription": "**template** (string) _optional_\n\nThe template file(s) which render the manifest to be tested, default to the list of template file defined in templates of suite file, unless template is defined in the assertion(s)."
+          },
+          "templates": {
+            "$ref": "#/definitions/templates"
+          },
+          "documentIndex": {
+            "$ref": "#/definitions/documentIndex"
+          },
+          "release": {
+            "$ref": "#/definitions/release"
+          },
+          "capabilities": {
+            "$ref": "#/definitions/capabilities"
+          },
+          "chart": {
+            "$ref": "#/definitions/chart"
+          },
+          "asserts": {
+            "type": "array",
+            "description": "The assertions to validate the rendered chart.",
+            "markdownDescription": "**asserts** (array<object>) _required_\n\nThe assertions to validate the rendered chart.",
+            "items": {
+              "type": "object",
+              "properties": {
+                "equal": true,
+                "equalRaw": true,
+                "notEqual": true,
+                "notEqualRaw": true,
+                "matchRegex": true,
+                "matchRegexRaw": true,
+                "notMatchRegex": true,
+                "notMatchRegexRaw": true,
+                "contains": true,
+                "notContains": true,
+                "isNull": true,
+                "isNotNull": true,
+                "isEmpty": true,
+                "isNotEmpty": true,
+                "isKind": true,
+                "isAPIVersion": true,
+                "hasDocuments": true,
+                "containsDocument": true,
+                "matchSnapshot": true,
+                "matchSnapshotRaw": true,
+                "failedTemplate": true,
+                "not": {
+                  "type": "boolean",
+                  "description": "Set to true to assert contrarily, default to false.",
+                  "markdownDescription": "**not** (boolean) _optional_\n\nSet to `true` to assert contrarily, default to `false`."
+                },
+                "template": {
+                  "type": "string",
+                  "description": "The template file which render the manifest to be asserted, default to the list of template files defined in templates of the suite file, unless the template is in the testjob.",
+                  "markdownDescription": "**template** (string) _optional_\n\nThe template file which render the manifest to be asserted, default to the list of template files defined in `templates` of the suite file, unless the template is in the testjob."
+                },
+                "documentIndex": {
+                  "$ref": "#/definitions/documentIndex"
+                }
+              },
+              "additionalProperties": false,
+              "oneOf": [
```

Comment:
This is awesome !
Sadly with latest IJ 2021.3.2 I'm getting an error "Schema validation: Validates to more than one variant" on each assert element, even though auto-completion works fine.
Seems confused by the oneOf, could be https://youtrack.jetbrains.com/issue/IDEA-236928, I see other tickets about similar error have been fixed though.

![image](https://user-images.githubusercontent.com/58322910/154020333-e8e04c73-cdcb-40a3-bc57-54b0aee2984b.png)

![image](https://user-images.githubusercontent.com/58322910/154020469-8c71b079-b364-4aca-b48b-9c28373f2861.png)

Reply:
Oh, that is too bad :/ I tested mostly on VSCode, so I did not catch that IntelliJ is not properly visualizing the `oneOf` keyword. I checked the [draft-07](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.7.3) again and the keyword seems to exist in that version ...

However, thanks for testing that and providing screenshots of the look-and-feel in IntelliJ! :-) I hope this gets fixed soonish ... It would also be possible to change the schema and avoid usage of `oneOf`, but that would have a negative impact on the code completion (since different kind of assertions are mutually exclusive).